### PR TITLE
Chore: All tests under pkg/tests should be integration tests

### PR DESCRIPTION
**Pull request by marefr**
Based on my https://github.com/grafana/grafana/pull/58268#issuecomment-1317302889. I strongly believe all the tests under pkg/tests should be marked as integration tests since they're starting up a real Grafana instance for each test.

Earlier durations (based on https://drone.grafana.net/grafana/grafana/92283/1/7):
step | duration |
------ | ------------- |
test-backend | 4m7s |
test-backend-integration | 5m12s |
postgres-integration-tests | 16m7s |
mysql-integration-tests | 16m6s |

Current durations (based on these changes):
step | duration | diff |
------ | ------------- | ----- |
test-backend | 1m14s | -30% |
test-backend-integration | 4m53s | -6,4% |
postgres-integration-tests | 18m20s | +13,7% |
mysql-integration-tests | 18m20s | +13,8% |

Might be some more optimizations to be done of the unit tests. Should the following tests perhaps be integration tests rather since it takes half a minute, doesn't really feels like unit tests? Or if unit can it be optimized to run faster?
- github.com/grafana/grafana/pkg/services/ngalert/sender  27.340s 

This one just noticed it takes over 5s, can it be optimized to run faster?
- github.com/grafana/grafana/pkg/services/ngalert/notifier  5.852s

### CODE REVIEW

**Comment by joeblubaugh** [comment]
> github.com/grafana/grafana/pkg/services/ngalert/notifier 5.852s

This one can be shortened by about 3 seconds by adjusting a few test parameters. Getting it down below that is difficult because of the dependent code from prometheus that the test depends on. These are in TestSilenceCleanup, where:
```
// Expiring soon.
makeSilence("", "tests", dt(now.Add(-5*time.Hour)), dt(now.Add(5*time.Second)), matchers),
// Expiring *very* soon
makeSilence("", "tests", dt(now.Add(-5*time.Hour)), dt(now.Add(2*time.Second)), matchers),
```

can be changed to 
```
// Expiring soon.
makeSilence("", "tests", dt(now.Add(-5*time.Hour)), dt(now.Add(2*time.Second)), matchers),
// Expiring *very* soon
makeSilence("", "tests", dt(now.Add(-5*time.Hour)), dt(now.Add(1*time.Second)), matchers),
```

> github.com/grafana/grafana/pkg/services/ngalert/sender 27.340s

I would feel comfortable marking a number of these as integration tests, because they spin up a local HTTP server that is used in tests. Any test with the name "TestSendingToExternal..." can be marked as an integration test.

File: pkg/tests/api/alerting/api_admin_configuration_test.go [hunk 1]
```diff
@@ -25,7 +25,11 @@ import (
 	"github.com/grafana/grafana/pkg/tests/testinfra"
 )
 
-func TestAdminConfiguration_SendingToExternalAlertmanagers(t *testing.T) {
+func TestIntegrationAdminConfiguration_SendingToExternalAlertmanagers(t *testing.T) {
```
Comment (papagian) [inline pkg/tests/api/alerting/api_admin_configuration_test.go:28]: If we rename them including the `Integration` they will run against [MySQL integration tests](https://github.com/grafana/grafana/blob/5bb99775bb995436e9fcf723ed1e622b685a9f1f/.drone.yml#L1725) (same for PostgreSQL integration tests); I'm not sure if we want that.

**Review by papagian — COMMENTED** [review]
I agree that they are not unit tests and they should be skipped but I'm not sure if we want to run them as well against other backends than SQLite

File: pkg/tests/api/alerting/api_admin_configuration_test.go (hunk 1 above)
Comment (marefr) [reply pkg/tests/api/alerting/api_admin_configuration_test.go:28]: Oh good catch :+1: It seems like it added like 5-6 min to the integration tests duration. For now I could skip them if not sqlite is used.

Comment (marefr) [reply pkg/tests/api/alerting/api_admin_configuration_test.go:28]: Pushed a suggestion to handle this. Not sure it make sense, WDYT?

**Review by papagian — APPROVED** [review]

**Comment by marefr** [comment]
@joeblubaugh thanks. 

> I would feel comfortable marking a number of these as integration tests, because they spin up a local HTTP server that is used in tests. Any test with the name "TestSendingToExternal..." can be marked as an integration test.

Maybe something you can plan for in upcoming PR?

Found another one that takes quite some time. Possible to speed up these?

sqlite: 
github.com/grafana/grafana/pkg/services/ngalert/store	246.906s
postgres: 
github.com/grafana/grafana/pkg/services/ngalert/store	252.245s
mysql:
github.com/grafana/grafana/pkg/services/ngalert/store	249.596s

Seems to be `TestIntegrationAlertInstanceBulkWrite` taking most of the time (251.701s locally). Not sure what our approach is for things like these which are more like benchmark tests or "long" integration tests. We do have the sense of short tests and integration tests, feels to me we might want to introduce long integration tests that perhaps not runs on PR builds? Would changing this to a benchmark make sense? Don't think we run benchmarks in CI so might be something we need to look into eventually as well.

**Comment by joeblubaugh** [comment]
@marefr here's a PR to reduce unit and integration test time: https://github.com/grafana/grafana/pull/60067

**Comment by marefr** [comment]
@papagian @joeblubaugh I added durations to the PR description. Seems to me that I might have done something wrong, e.g. test-backend-integration takes shorter time after my changes :thinking: Would like a reality check, WDYT?

**Pull request merged**